Rockchip: initial support

Kwiboo opened the pull request:
This PR adds initial support for Rockchip SoCs (video HW decoding not included).

~~This include commits from #2100 and #2108 and will be rebased once they have been merged into master.~~
~~There is also a temporary commit to fix a aarch64 build issue caused by `pfcd_hevc_optimisations.patch` that will be dropped once a proper solution is merged into master.~~

There are some issues to resolve / work on before this is ready to be merged:
* [x] Swtich to mainline u-boot ~~and drop the project specific package~~
* [x] Drop the use of device specific mkimage scripts
* [x] LED control is no longer working using the latest RK 4.4 kernel
* [x] Update and combine linux patches
* [x] ~~Possible WiFi issue on 64/32-bit build~~

Big thanks to @LongChair, @omegamoon, @Raybuntu and Rockchip engineers for great collaboration working on this.

LongChair commented:
Awesome work @Kwiboo ! This has been a long time work :) also special thanks to @yanghanxing and his team for the great support he's been bringing us !

Raybuntu commented:
Great job @Kwiboo!

CvH commented on packages/graphics/mali-rockchip/package.mk:
```diff
@@ -0,0 +1,110 @@
+PKG_SHORTDESC="mali-rockchip: OpenGL ES user-space binary for the ARM Mali GPU family"
+PKG_LONGDESC="mali-rockchip: OpenGL ES user-space binary for the ARM Mali GPU family"
+
+PKG_IS_ADDON="no"
```
rm `PKG_IS_ADDON="no"` and change to
```
PKG_SHORTDESC=""
PKG_LONGDESC=""
PKG_AUTORECONF="no"
```

CvH commented on packages/lang/gcc-linaro-aarch64-linux-gnu/package.mk:
```diff
@@ -0,0 +1,41 @@
+################################################################################
+#      This file is part of LibreELEC - https://libreelec.tv
+#      Copyright (C) 2016 Team LibreELEC
```
2016-present

CvH commented on packages/lang/gcc-linaro-aarch64-linux-gnu/package.mk:
```diff
@@ -0,0 +1,41 @@
+PKG_LONGDESC=""
+
+PKG_IS_ADDON="no"
+PKG_AUTORECONF="no"
```
rm `PKG_IS_ADDON="no"`
```
PKG_SHORTDESC=""
PKG_LONGDESC=""
PKG_AUTORECONF="no"
```

Raybuntu replied:
I've fixed those in my PR so a rebase should be sufficient :)

CvH commented on packages/multimedia/rkmpp/package.mk:
```diff
@@ -0,0 +1,51 @@
+PKG_LONGDESC="rkmpp: Rockchip Media Process Platform (MPP) module"
+
+PKG_IS_ADDON="no"
+PKG_AUTORECONF="no"
```
rm `PKG_IS_ADDON="no"`
```
PKG_SHORTDESC=""
PKG_LONGDESC=""
PKG_AUTORECONF="no"
```

CvH commented on packages/multimedia/rkmpp/package.mk:
```diff
@@ -0,0 +1,51 @@
+PKG_AUTORECONF="no"
+
+if [ "$UBOOT_SOC" = "rk3328" ]; then
+  ENABLE_VP9D="ON"
```
`ENABLE_VP9D` should be `PKG_ENABLE_VP9D` as we try to rework every custome var to PKG_... (same change `DENABLE_VP9D=$ENABLE_VP9D`)

CvH commented on packages/network/rfkill/package.mk:
```diff
@@ -0,0 +1,32 @@
+PKG_LONGDESC="rfkill is a small userspace tool to query the state of the rfkill switches, buttons and subsystem interfaces."
+
+PKG_IS_ADDON="no"
+PKG_AUTORECONF="no"
```
rm `PKG_IS_ADDON="no"`
```
PKG_SHORTDESC=""
PKG_LONGDESC=""
PKG_AUTORECONF="no"
```

CvH commented on packages/tools/rkbin/package.mk:
```diff
@@ -0,0 +1,40 @@
+PKG_LONGDESC="rkbin: Rockchip Firmware and Tool Binaries"
+
+PKG_IS_ADDON="no"
+PKG_AUTORECONF="no"
```
rm `PKG_IS_ADDON="no"`
```
PKG_SHORTDESC=""
PKG_LONGDESC=""
PKG_AUTORECONF="no"
```

CvH commented on projects/Rockchip/devices/ROCK64/scripts/mkimage:
```diff
@@ -0,0 +1,48 @@
+################################################################################
+#      This file is part of LibreELEC - https://LibreELEC.tv
```
`https://LibreELEC.tv` -> `https://libreelec.tv`

CvH commented on projects/Rockchip/devices/ROCK64/scripts/mkimage:
```diff
@@ -0,0 +1,48 @@
+################################################################################
+#      This file is part of LibreELEC - https://LibreELEC.tv
+#      Copyright (C) 2017 Team LibreELEC
```
2017-present

CvH commented on projects/Rockchip/devices/TinkerBoard/scripts/mkimage:
```diff
@@ -0,0 +1,40 @@
+################################################################################
+#      This file is part of LibreELEC - https://LibreELEC.tv
+#      Copyright (C) 2017 Team LibreELEC
```
2017-present

CvH commented on projects/Rockchip/devices/TinkerBoard/scripts/mkimage:
```diff
@@ -0,0 +1,40 @@
+################################################################################
+#      This file is part of LibreELEC - https://LibreELEC.tv
```
https://libreelec.tv

CvH commented on projects/Rockchip/packages/rtl8723b_bt/package.mk:
```diff
@@ -0,0 +1,47 @@
+################################################################################
+
+PKG_NAME="rtl8723b_bt"
+PKG_VERSION=""
```
pls gave it some version or we get `0.0invaild` instead

CvH commented on projects/Rockchip/packages/rtl8723b_bt/package.mk:
```diff
@@ -0,0 +1,47 @@
+PKG_LONGDESC="rtl8723b_bt firmware"
+
+PKG_IS_ADDON="no"
+PKG_AUTORECONF="no"
```
rm `PKG_IS_ADDON="no"` and change to
```
PKG_SHORTDESC=""
PKG_LONGDESC=""
PKG_AUTORECONF="no"
```

Kwiboo commented:
Rebased on latest master and updated #2100 commits, fixes for @CvH's comments have been squashed.
Also dropped a commit no longer needed after a small update to the devices specific `mkimage` scripts.

Kwiboo commented:
Updated to latest RK 4.4 kernel, u-boot v2017.11-rc3 and linux patches is now merged files matching my [linux-rockchip topic branches](https://github.com/Kwiboo/linux-rockchip/branches/all).

This should be ready to merge after a final boot test of new clean builds.

Kwiboo commented:
Clean builds boots and starts kodi, test images:

[LibreELEC-ROCK64.aarch64-9.0-devel-20171102001520-r27249-g4f038a0.img.gz](https://openpht.blob.core.windows.net/kwiboo/LibreELEC-ROCK64.aarch64-9.0-devel-20171102001520-r27249-g4f038a0.img.gz)
[LibreELEC-ROCK64.arm-9.0-devel-20171102034700-r27249-g4f038a0.img.gz](https://openpht.blob.core.windows.net/kwiboo/LibreELEC-ROCK64.arm-9.0-devel-20171102034700-r27249-g4f038a0.img.gz)
[LibreELEC-TinkerBoard.arm-9.0-devel-20171102020010-r27249-g4f038a0.img.gz](https://openpht.blob.core.windows.net/kwiboo/LibreELEC-TinkerBoard.arm-9.0-devel-20171102020010-r27249-g4f038a0.img.gz)

Kwiboo commented:
Thanks! :+1:

chewitt commented:
@Kwiboo nice work! .. and big thanks to all the others that helped move this forwards :)
